packages/components/src/internal/components/pipeline/utils.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import { AppURL } from '../../url/AppURL';
+import { App, AssayUploadResultModel } from '../../../index';
+
+export function getPipelineLinkMsg(response: AssayUploadResultModel): JSX.Element {
```

Comment:
Curious why you prefer `JSX.Element` over `ReactNode`? We've normally utilized `ReactNode` as it is slightly more generic and matches what react says `render()` returns. If you don't have a preference, then I'd consider switching usages of `JSX.Element` to `ReactNode` throughout this PR.